File: JUNGOL/Beginner/#1339 문자삼각형2/문자삼각형2.py
n = int(input())

if n<1 or n>100 or n%2==0:
    print("INPUT ERROR")
else:
    tri = [[0 for i in range(n)] for i in range(n)]

    c = ord('A')
    a,b = n//2,1
    for i in range((n+1)//2):
        for j in range(a, a+b):
            tri[j][n//2-i] = c
            c += 1
            if c > ord('Z'): c = ord('A')

        a,b = a-1,b+2

    for t in tri:
        for ti in t:
            if ti == 0: continue
            print(chr(ti), end=' ')
        print()

File: JUNGOL/Beginner/#1339 문자삼각형2/문자삼각형2.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
    int i,j;

    int n; cin>>n;
    if(n<1 || n>100 || !(n%2)){
        cout<<"INPUT ERROR\n";
        return 0;
    }

    vector<vector<char>> tri;
    vector<char> v(n/2+1, ' ');
    for(i=0; i<n; i++)
        tri.push_back(v);

    char c = 'A';
    int a=n/2, b=1;
    for(i=0; i<(n+1)/2; i++){
        for(j=a; j<a+b; j++){
            tri[j][n/2-i] = c++;
            if(c > 'Z') c = 'A';
        }

        a--; b+=2;
    }

    for(vector<char> t : tri){
        for(char ti : t) printf("%c ", ti);
        printf("\n");
    }
}
